src/app/loadout-drawer/loadout-apply.ts

```diff
@@ -735,58 +750,84 @@ function equipMods(
     const existingModsBySocketType = new Map<DimSocket, DimPlug[]>()
     */
 
-    const modSockets = item.sockets.allSockets.filter(isArmorModSocket);
-
-    const modsToApply = [...modsForItem];
+    const [removals, desireds] = _.partition(
+      assignmentInstructions,
+      // eslint-disable-next-line radar/no-identical-functions
+      (assignment) =>
+        assignment.mod.hash ===
+        getSocketByIndex(item.sockets!, assignment.socketIndex)!.socketDefinition
+          .singleInitialItemHash
+    );
```

Comment:
Just double checking that we are sure this keeps the ordering, its just that its split across the partition?